package/kedro_viz/api/graphql.py

```diff
@@ -37,51 +37,63 @@ class JSONObject(dict):
     )
 
 
-def format_run(run_id: str, run_blob: Dict) -> Run:
+def format_run(
+    run_id: str, run_blob: Dict, user_run_details: Optional[UserRunDetailsModel]
+) -> Run:
     """Convert blob data in the correct Run format.
     Args:
         run_id: ID of the run to fetch
         run_blob: JSON blob of run metadata and tracking data
+        user_run_details: The user run details associated with this run
     Returns:
         Run object
     """
-    session = data_access_manager.db_session
     git_data = run_blob.get("git")
-    user_details = (
-        session.query(UserRunDetailsModel)
-        .filter(UserRunDetailsModel.run_id == run_id)
-        .scalar()
```

Comment:
`format_run` is serialisation logic. It shouldn't have data loading logic. Make it easier to unit test.